### PR TITLE
Migrate from hamcrest to assertj #222

### PR DESCRIPTION
**Pull request by ishanarora04**

### CODE REVIEW

**Comment by ishanarora04** [comment]
This is open for review.

File: greenmail-core/src/test/java/com/icegreen/greenmail/test/GreenMailUtilTest.java [hunk 1]
```diff
@@ -61,24 +59,23 @@ public void testGetEmptyBodyAndHeader() throws Exception {
                 MimeMultipart mp = (MimeMultipart) retriever.getMessages(to)[0].getContent();
                 BodyPart bp;
                 bp = mp.getBodyPart(0);
-                assertEquals(body, GreenMailUtil.getBody(bp).trim());
-                assertEquals(
+                assertThat(GreenMailUtil.getBody(bp).trim()).isEqualTo(body);
+                assertThat(
```
Comment (marcelmay) [inline greenmail-core/src/test/java/com/icegreen/greenmail/test/GreenMailUtilTest.java:63]: Switch 1st/2nd arg?

File: greenmail-core/src/test/java/com/icegreen/greenmail/test/GreenMailUtilTest.java [hunk 2]
```diff
@@ -61,24 +59,23 @@ public void testGetEmptyBodyAndHeader() throws Exception {
...
                         "Content-Type: text/plain; charset=us-ascii\r\n" +
-                        "Content-Transfer-Encoding: 7bit",
+                        "Content-Transfer-Encoding: 7bit").isEqualTo(
                         GreenMailUtil.getHeaders(bp).trim());
 
                 bp = mp.getBodyPart(1);
-                assertEquals("AAEC", GreenMailUtil.getBody(bp).trim());
-                assertEquals(
+                assertThat(GreenMailUtil.getBody(bp).trim()).isEqualTo("AAEC");
+                assertThat(
```
Comment (marcelmay) [inline greenmail-core/src/test/java/com/icegreen/greenmail/test/GreenMailUtilTest.java:70]: Switch 1st/2nd arg?

File: pom.xml [hunk 3]
```diff
@@ -490,9 +490,9 @@
         <version>4.2</version>
       </dependency>
       <dependency>
-        <groupId>org.hamcrest</groupId>
-        <artifactId>hamcrest</artifactId>
-        <version>2.2</version>
+         <groupId>org.assertj</groupId>
+         <artifactId>assertj-core</artifactId>
+         <version>3.11.1</version>
```
Comment (marcelmay) [inline pom.xml:495]: Use latest version 3.16.1 ?

**Comment by marcelmay** [comment]
@ishanarora04 , thx a lot for the PR. That was very fast. I have minor comments, which I can also fix.

File: greenmail-core/src/test/java/com/icegreen/greenmail/test/commands/POP3CommandTest.java [hunk 4]
```diff
@@ -41,70 +37,70 @@ public void setUp() {
...
             PrintStream printStream = new PrintStream(socket.getOutputStream());
             final BufferedReader reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
 
             greenMail.getManagers().getUserManager().setAuthRequired(true);
 
-            assertThat(reader.readLine(), is(startsWith("+OK POP3 GreenMail Server v")));
+            assertThat(reader.readLine()).startsWith("+OK POP3 GreenMail Server v");
             printStream.print("USER blar@blar.com" + CRLF);
-            assertThat(reader.readLine(), is(not(equalTo("+OK"))));
+            assertThat(reader.readLine()).isEqualTo("+OK");
```
Comment (marcelmay) [inline greenmail-core/src/test/java/com/icegreen/greenmail/test/commands/POP3CommandTest.java:103]: isNotEqualTo

**Comment by marcelmay** [comment]
greenmail-standalone/pom.xml would also need the assertj dependency:

    <dependency>                                                                                                                    
        <groupId>org.assertj</groupId>                                                                                               
        <artifactId>assertj-core</artifactId>                                                                                        
        <scope>test</scope>                                                                                                          
    </dependency>

**Comment by ishanarora04** [comment]
I am fixing the comments. Almost ready to commit .

**Comment by ishanarora04** [comment]
Any other suggestions

**Pull request merged**